### DISCUSSION TITLE
DAL - Add option during update / upsert that removes all non provided relation entities, e.g. product visibility (similar to doctrine's orphan removal)

### DISCUSSION THREAD

**Discussion by AndreasA** [shopware/platform#3149] [RFC / Proposal]
**Effort**:  middle (probably)

### Description: 

I had the requirement quite often to update entities (upsert or update) with relations, where if the related entity was not provided, it should bedeleted, so e.g. one can easily create new relations and remove unused ones (and maybe update related entities if necessary).
Should probably only be done for associations where an `array` is provided, if nothing exists then no update should happen. And it should be optional and not enabled by default.

This would be similar to doctrine's orphan removal.

It would also be helpful when e.g. updating a sales channel through code and adjusting its assigned language list, provided language relations are kept or created and others are removed. Especially as due to the requirement for keeping the system language relation, it isn't just possible to delete all relations and then add new ones.

### Benefits

see description.

### Break strategy

As it is an optional setting, it wouldn't break anything as long as it isn't used. Either by adding an optional parameter to the methods (which as iti s optional doesn't provide an issue on overrides) or by adding an additional state to the context object.

**Comment by AndreasA** [comment]
might be related to https://github.com/shopware/platform/issues/3125

**Comment by OliverSkroblin** [comment]
Hey @AndreasA,

i just implemented something like this during the last Hockethon of the Kellerkinder. It is also already merged:

https://github.com/shopware/platform/commit/30e574c7499e102e162a521dd3348d1756b739c0

It's not exactly what you requested, but it goes in the same direction.

**Comment by OliverSkroblin** [comment]
Hey, 

I have closed the discussion because with this proposal and the linked commit a similar behavior can be realized.

https://github.com/shopware/platform/discussions/3150

I hope this is okay for you
